internal/net/udp/packet_conn.go

```diff
@@ -0,0 +1,399 @@
+// SPDX-FileCopyrightText: 2023 The Pion community <https://pion.ly>
+// SPDX-License-Identifier: MIT
+
+// Package udp implements DTLS specific UDP networking primitives.
+// NOTE: this package is an adaption of pion/transport/udp. If possible, the
+// updates made in this repository will be reflected back upstream. If not, it
+// is likely that this will be moved to a public package in this repository.
```

Comment:
I like the idea of making the history more understandable, but I am weary of doing so by trying to describe each of the differences in comments because that feels like a recipe for becoming stale over time if more changes are made in this package. I would recommend providing a permalink to the commit in `pion/transport` at which this was copied and changed, such that folks can use `git diff` to see the exact changes that were made, as well as any subsequent changes that were made here or in `pion/transport`. That being said, I think expanding the current comment to describe _why_ this was "forked" would be useful. What do you think about that strategy?

Reply:
yeah that's very reasonable, a summary is fine!